packages/cosmwasm-stargate/src/queries/wasm.ts

```diff
@@ -67,54 +63,80 @@ export interface WasmExtension {
 }
 
 export function setupWasmExtension(base: QueryClient): WasmExtension {
-  const queryService = Query.create((method: any, requestData, callback) => {
-    const path = `/cosmwasm.wasm.v1beta1.Query/${method.name}`;
-    base
-      .queryUnverified(path, requestData)
-      .then((response) => callback(null, response))
-      .catch((error) => callback(error));
-  });
+  const rpc = createRpc(base);
+  // Use this service to get easy typed access to query methods
+  // This cannot be used for proof verification
+  const queryService = new QueryClientImpl(rpc);
+
   return {
     unverified: {
       wasm: {
         listCodeInfo: async (paginationKey?: Uint8Array) => {
-          const request = paginationKey ? { pagination: { key: paginationKey } } : {};
-          return queryService.codes(request);
+          const request = {
+            pagination: {
```

Comment:
When `paginationKey` is unset, we can (and probably should) leave the whole `pagination` value `undefined`